### PR TITLE
PROD-267: Color Audit

### PR DESCRIPTION
**Pull request by neel-ds**
### Description
- Make necessary changes in shadcn components and switch to new button wherever possible
- Remove hard coded colors wherever possible to align with purple/gray
- Stick to color primitives and semantic colors
### What are the steps to test that this code is working?
Best to try the whole app and review consistent UI
### Screen shots or recordings for UI changes
Screenshot 2024-09-28 at 12 53 33 AM
Screenshot 2024-09-28 at 12 54 02 AM

### CODE REVIEW

File: app/providers/RevealProvider.tsx [hunk 1]
```diff
@@ -297,7 +297,7 @@ export function RevealContextProvider({
                       closeRevealModal();
                     }
                   }}
-                  className="absolute top-5 right-6 border-none w-max !p-0 z-50"
+                  className="!absolute !top-3 !right-6 !border-none !w-max !p-0 z-50"
```
Comment (yash431garg) [inline app/providers/RevealProvider.tsx:300]: Does adding exclamation mark behind every property is necessary here?

File: tailwind.config.ts
Comment (yash431garg) [inline tailwind.config.ts:1]: Wouldn't using "destructive" instead of "red" be confusing?

File: app/providers/RevealProvider.tsx (hunk 1 above)
Comment (neel-ds) [reply app/providers/RevealProvider.tsx:300]: To override the button variant, it's necessary.

File: tailwind.config.ts
Comment (neel-ds) [reply tailwind.config.ts:1]: According to Figma color primitives, "destructive" should be used.

**Review by yash431garg — APPROVED** [review]

**Pull request merged**